test/spec/controllers/eventRegistrations.spec.js

```diff
@@ -150,4 +348,223 @@ describe('Controller: eventRegistrations', function () {
+        .respond(204, '');
+
+      const registrant = {
+        ...testData.registration.registrants[0],
+        withdrawn: true,
+        withdrawnTimestamp: '2023-01-01T00:00:00.000Z',
+      };
+      scope.withdrawRegistrant(registrant, false);
+
+      expect(registrant.withdrawn).toBe(false);
+      expect(scope.loadingMsg).toBe('Reinstating Test');
+
+      $httpBackend.flush();
+
+      expect(scope.loadingMsg).toBe('');
+    });
+
+    it('reverts changes to withdrawn on failure', () => {
+      $httpBackend
+        .expectPUT(/^registrants\/.+$/, (data) => JSON.parse(data).withdrawn)
+        .respond(500, '');
+
+      const registrant = {
+        ...testData.registration.registrants[0],
+        withdrawn: false,
+        withdrawnTimestamp: null,
+      };
+      scope.withdrawRegistrant(registrant, true);
+      $httpBackend.flush();
+
+      expect(registrant.withdrawn).toBe(false);
+    });
+  });
+
+  describe('checkInRegistrant', () => {
+    it('checks in registrant', () => {
+      $httpBackend
+        .expectPUT(/^registrants\/.+$/, (data) =>
+          angular.isString(JSON.parse(data).checkedInTimestamp),
+        )
+        .respond(204, '');
+
+      const registrant = {
+        ...testData.registration.registrants[0],
+        checkedInTimestamp: null,
+      };
+      scope.checkInRegistrant(registrant, true);
+
+      expect(registrant.checkedInTimestamp).not.toBeNull();
+      expect(scope.loadingMsg).toBe('Checking in Test');
+
+      $httpBackend.flush();
+
+      expect(scope.loadingMsg).toBe('');
+    });
+
+    it('removes check in from registrant', () => {
+      $httpBackend
+        .expectPUT(
+          /^registrants\/.+$/,
+          (data) => JSON.parse(data).checkedInTimestamp === null,
+        )
+        .respond(204, '');
+
+      const registrant = {
+        ...testData.registration.registrants[0],
+        checkedInTimestamp: '2023-01-01T00:00:00.000Z',
+      };
+      scope.checkInRegistrant(registrant, false);
+
+      expect(registrant.checkedInTimestamp).toBeNull();
+      expect(scope.loadingMsg).toBe('Removing check-in for Test');
+
+      $httpBackend.flush();
+
+      expect(scope.loadingMsg).toBe('');
+    });
+
+    it('reverts changes to checkedInTimestamp on failure', () => {
+      $httpBackend
+        .expectPUT(/^registrants\/.+$/, (data) =>
+          angular.isString(JSON.parse(data).checkedInTimestamp),
+        )
+        .respond(500, '');
+
+      const registrant = {
+        ...testData.registration.registrants[0],
+        checkedInTimestamp: null,
+      };
+      scope.checkInRegistrant(registrant, true);
+
+      expect(registrant.checkedInTimestamp).not.toBeNull();
+
+      $httpBackend.flush();
+
+      expect(registrant.checkedInTimestamp).toBeNull();
+    });
+  });
+
+  describe('deleteRegistrant', () => {
+    it('deletes a registrant in a group', () => {
+      $httpBackend.expectDELETE(/^registrants\/.+$/).respond(204, '');
```

Comment:
Can we do a test where there are more than just one registrant in a group when one is deleted? This is linked to my comment about the error when deleting a registrant in the modal.

Reply:
Maybe this is doing that. But the error I had is around the same code

Reply:
Yes, this tests removing one registrant from a group with multiple registrants. I reverted the change to the controller, and that should fix the error.